tools/toolchains/arm.py

```diff
@@ -554,7 +570,17 @@ def __init__(self, target, *args, **kwargs):
                     "ARM/ARMC6 compiler support is required for ARMC6 build"
                 )
 
-        if getattr(target, "default_toolchain", "ARMC6") == "uARM":
+        if  getattr(target, "default_lib", "std") == "small":
+            if (
+                hasattr(target, "supported_c_libs") == False
+                or "arm" not in target.supported_c_libs
+                or "small" not in target.supported_c_libs["arm"]
+            ):
+                raise NotSupportedException(
+                  "target.default_lib small not supported for this target."
```

Comment:
Write "small C library not supported for this target" so you don't have to change the message if/when default_lib becomes "c_lib".